app/src/main/java/org/oppia/app/options/OptionsFragment.kt

```diff
@@ -0,0 +1,185 @@
+          AppLanguageActivity.createAppLanguageActivityIntent(
+            requireContext(),
+            getString(R.string.key_app_language),
+            appLanguagePref.summary.toString()
+          ), 2
+        )
+        return true
+      }
+    }
+
+    val defaultAudioPref = findPreference<Preference>(getString(R.string.key_default_audio))
+    defaultAudioPref.summary = audioLanguage
+    defaultAudioPref.onPreferenceClickListener = object : Preference.OnPreferenceClickListener {
+      override fun onPreferenceClick(preference: Preference): Boolean {
+        startActivityForResult(
+          DefaultAudioActivity.createDefaultAudioActivityIntent(
+            requireContext(),
+            getString(R.string.key_default_audio),
+            defaultAudioPref.summary.toString()
+          ), 3
+        )
+        return true
+      }
+    }
+  }
+
+  private fun bindPreferenceSummaryToValue(typeOfValue: String, preference: Preference) {
+    preference.onPreferenceChangeListener = bindPreferenceSummaryToValueListener
+
+    bindPreferenceSummaryToValueListener.onPreferenceChange(
+      preference, PreferenceManager
+        .getDefaultSharedPreferences(preference.context)
+        .getString(preference.key, typeOfValue)
+    )
+  }
+
+  private val bindPreferenceSummaryToValueListener =
+    Preference.OnPreferenceChangeListener { preference, newValue ->
+      val stringValue = newValue.toString()
+      when {
+        preference.key == getString(R.string.key_story_text_size) -> // update the changed Text size  to summary filed
+        {
+          preference.summary = stringValue
+          when (stringValue) {
+            "Small" -> {
+              profileManagementController.updateStoryTextSize(profileId, 16f)
+            }
+            "Medium" -> {
+              profileManagementController.updateStoryTextSize(profileId, 18f)
+            }
+            "Large" -> {
+              profileManagementController.updateStoryTextSize(profileId, 20f)
+            }
+            "Extra Large" -> {
+              profileManagementController.updateStoryTextSize(profileId, 22f)
+            }
+          }
+        }
+        preference.key == getString(R.string.key_app_language) -> // update the changed language  to summary filed
+        {
+          preference.summary = stringValue
+          profileManagementController.updateAppLanguage(profileId, stringValue)
+        }
+        preference.key == getString(R.string.key_default_audio) -> // update the changed audio language  to summary filed
+        {
+          preference.summary = stringValue
+          profileManagementController.updateAudioLanguage(profileId, stringValue)
+        }
+      }
+      true
+    }
+
+  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    // Check which request we're responding to
+    when (requestCode) {
+      1 -> {
+        val textSize = data!!.getStringExtra("MESSAGE") as String
```

Comment:
Why are these keys same, they should be different with proper context like `KEY_TEXT_SIZE`, also, create then as constants and use those keys wherever needed.

Reply:
Created constants to pass through intents.